cmd/kubeadm/app/cmd/phases/upgrade/node/kubeletconfig.go

```diff
@@ -78,9 +78,13 @@ func runKubeletConfigPhase() func(c workflow.RunData) error {
 		}
 		src := filepath.Join(kubeletDir, constants.KubeletConfigurationFileName)
 		dest := filepath.Join(backupDir, constants.KubeletConfigurationFileName)
-		fmt.Printf("[upgrade] backing up kubelet config file to %s\n", dest)
-		if err := os.Rename(src, dest); err != nil {
-			return errors.Wrap(err, "error backing up the kubelet config file")
+		if !dryRun {
+			fmt.Printf("[upgrade] backing up kubelet config file to %s\n", dest)
+			if err := os.Rename(src, dest); err != nil {
+				return errors.Wrap(err, "error backing up the kubelet config file")
+			}
+		} else {
+			fmt.Printf("[upgrade] Would back up kubelet config file to %s\n", dest)
 		}
 		// Store the kubelet component configuration.
 		if err = kubeletphase.WriteConfigToDisk(&cfg.ClusterConfiguration, kubeletDir, data.PatchesDir(), data.OutputWriter()); err != nil {
```

Comment:
Can you pls check this for dry-run as well?

Reply:
okay, I think the original logic is that write the config to a fake dir if it's dry run. If we agree that original logic is okay, there is no need to distinguish the dry-run or not for backing up and write config.

so, I think,
- we don't need this change and do all those thing in the fake dir if it is dry-run mode.
- don't create the fake dir (as we won't use it), and just print instead of action if it's dry-run mode.

Reply:
we can get away with not writing the backup file during dryrun and just print the message..."would..."

but if we are writing backup of manifest files during kubeadm upgrade dryrun we might also want to keep this kubelet backup writing during dryrun for consistency.

Reply:
I went through the relevant code, and confirm that
- the manifest files will be written to the /tmp dir if upgrading a node with dry-run mode.
- this is needed since kubeadm would print the content of the manifest or kubelet conf if it's dry-run mode, this requires that the config or manifest files should be written to the /tmp dir beforehand.

Reply:
right, i forgot about that.

still not clear to what is best, but if you have a pr in mind just send it please.